core/akka-runtime/src/test/scala/com/ing/baker/runtime/akka/actor/process_instance/ProcessInstanceSpec.scala

```diff
@@ -664,5 +676,284 @@ class ProcessInstanceSpec extends AkkaTestBase("ProcessInstanceSpec") with Scala
+
+      val eventsToFire: Seq[EventDescriptor] = Seq(
+        EventDescriptor(eventName, Seq.empty),
+        EventDescriptor(exhaustedEvent, Seq.empty)
+      )
+
+      val failureStrategy: InteractionFailureStrategy = FireEventAfterFailure(EventDescriptor(exhaustedEvent, Seq.empty))
+
+      val interactionTransition: InteractionTransition = InteractionTransition(
+        eventsToFire, eventsToFire,
+        Seq.empty,
+        "Name",
+        "Name",
+        Map.empty,
+        Option.empty,
+        failureStrategy, Map.empty, false)
+
+      val logMock = mock[DiagnosticLoggingAdapter]
+
+      val output: String = ProcessInstance.getOutputEventName(interactionTransition, logMock)
+
+      assert(output == eventName)
+    }
+
+    "Should correctly determine the OutputEventName for Delayed Transitions if RetryWithIncrementalBackoff is used with FireEvent" in new TestSequenceNet {
+      override val sequence = Seq(
+        transition() { _ => Added(1) })
+
+      val eventName = "originalEvent1"
+      val exhaustedEvent = "exhaustedEvent"
+
+      val eventsToFire: Seq[EventDescriptor] = Seq(
+        EventDescriptor(eventName, Seq.empty),
+        EventDescriptor(exhaustedEvent, Seq.empty)
+      )
+
+      val failureStrategy: InteractionFailureStrategy =
+        RetryWithIncrementalBackoff(Zero, 1.0, 1, Option.empty, Some(EventDescriptor(exhaustedEvent, Seq.empty)))
+
+      val interactionTransition: InteractionTransition = InteractionTransition(
+        eventsToFire, eventsToFire,
+        Seq.empty,
+        "Name",
+        "Name",
+        Map.empty,
+        Option.empty,
+        failureStrategy, Map.empty, false)
+
+      val logMock = mock[DiagnosticLoggingAdapter]
+
+      val output: String = ProcessInstance.getOutputEventName(interactionTransition, logMock)
+
+      assert(output == eventName)
+    }
+
+    "Should correctly determine the OutputEventName for Delayed Transitions if RetryWithIncrementalBackoff is used without FireEvent" in new TestSequenceNet {
+      override val sequence = Seq(
+        transition() { _ => Added(1) })
+
+      val eventName = "originalEvent1"
+      val exhaustedEvent = "exhaustedEvent"
+
+      val eventsToFire: Seq[EventDescriptor] = Seq(
+        EventDescriptor(eventName, Seq.empty)
+      )
+
+      val failureStrategy: InteractionFailureStrategy =
+        RetryWithIncrementalBackoff(Zero, 1.0, 1, Option.empty, Option.empty)
+
+      val interactionTransition: InteractionTransition = InteractionTransition(
+        eventsToFire, eventsToFire,
+        Seq.empty,
+        "Name",
+        "Name",
+        Map.empty,
+        Option.empty,
+        failureStrategy, Map.empty, false)
+
+      val logMock = mock[DiagnosticLoggingAdapter]
+
+      val output: String = ProcessInstance.getOutputEventName(interactionTransition, logMock)
+
+      assert(output == eventName)
+    }
+
+    "Should get correct wait time from state" in new TestSequenceNet {
```

Comment:
Maybe we should also add a test for Java Duration?

Reply:
This test was for Java Duration but was missing the Scala FiniteDuration. Made separate tests for both.